packages/resolvers/glob/src/GlobResolver.js

```diff
@@ -60,73 +60,114 @@ export default (new Resolver({
     let invalidateOnFileCreate = [];
     let invalidateOnFileChange = new Set();
 
-    // if the specifier does not start with /, ~, or . then it's not a path but package-ish - we resolve
-    // the package first, and then append the rest of the path
-    if (!/^[/~.]/.test(specifier)) {
-      // Globs are not paths - so they always use / (see https://github.com/micromatch/micromatch#backslashes)
-      let splitOn = specifier.indexOf('/');
-      if (specifier.charAt(0) === '@') {
-        splitOn = specifier.indexOf('/', splitOn + 1);
+    switch (specifier[0]) {
+      // Path specifier
+      case '.': {
+        specifier = path.resolve(path.dirname(sourceFile), specifier);
+        break;
       }
 
-      // Since we've already asserted earlier that there is a glob present, it shouldn't be
-      // possible for there to be only a package here without any other path parts (e.g. `import('pkg')`)
-      invariant(splitOn !== -1);
-
-      let pkg = specifier.substring(0, splitOn);
-      let rest = specifier.substring(splitOn + 1);
-
-      // This initialisation code is copied from the DefaultResolver
-      const resolver = new NodeResolver({
-        fs: options.inputFS,
-        projectRoot: options.projectRoot,
-        packageManager: options.shouldAutoInstall
-          ? options.packageManager
-          : undefined,
-        mode: options.mode,
-        logger,
-      });
-
-      let result;
-      try {
-        result = await resolver.resolve({
-          filename: pkg + '/package.json',
-          parent: dependency.resolveFrom,
-          specifierType: 'esm',
-          env: dependency.env,
-          sourcePath: dependency.sourcePath,
-        });
-      } catch (err) {
-        if (err instanceof ThrowableDiagnostic) {
-          // Return instead of throwing so we can provide invalidations.
-          return {
-            diagnostics: err.diagnostics,
-            invalidateOnFileCreate,
-            invalidateOnFileChange: [...invalidateOnFileChange],
-          };
-        } else {
-          throw err;
-        }
+      // Absolute path. Make the glob relative to the project root.
+      case '/': {
+        specifier = path.resolve(options.projectRoot, specifier.slice(1));
```

Comment:
On Windows, this will rewrite slashes in the specifier into backslashes:
```js
> path.posix.resolve("/src/app","dir") // default path.resolve on Unix
'/src/app/dir'
> path.win32.resolve("/src/app","dir") // default path.resolve on Windows
'\\src\\app\\dir'
```